src/seedu/addressbook/Main.java

```diff
@@ -1,160 +1,140 @@
 
-/* ==============NOTE TO STUDENTS======================================
- * This class header comment below is brief because details of how to
- * use this class are documented elsewhere.
- * ====================================================================
- */
 
 /**
  * Entry point of the address book application.
- * Contains command execution logic; initialises and uses the different components.
+ * Initializes the application and starts the interaction with the user.
  */
 public class Main {
 
     /**
-     * Default file path used if the user doesn't provide the file name.
+     * Version info of the program.
      */
-    public static final String DEFAULT_STORAGE_FILEPATH = "addressbook.txt";
+    public static final String VERSION = "AddessBook Level 2 - Version 1.0";
 
-    /**
-     * These message is not in the TextUi class because program launch and initialisation messages may be shown
-     * outside the given UI (should be logged at launch console), and needs to be shown before the UI is ready.
-     */
-    public static final String MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE = "Launch arguments format: [STORAGE_FILE_PATH]";
-    public static final String MESSAGE_INIT_FAILED = "Failed to initialise address book application. Exiting...";
+    private TextUi ui;
+    private StorageFile storage;
+    private AddressBook addressBook;
 
     /**
-     * Version info of the program.
+     * The list of person shown to the user most recently.
      */
-    public static final String VERSION = "AddessBook Level 1 - Version 1.0";
+    private List<? extends ReadOnlyPerson> lastShownList = null;
 
 
+    public static void main(String... launchArgs) {
+        new Main().run(launchArgs, System.in, System.out);
+    }
+
     /**
-     * Signals that the main application had a problem while initialising.
+     * Runs the program until termination.
      */
-    public static class MainInitialisationException extends Exception {}
-
-    private final TextUi ui;
-    private final Parser parser;
-    private final StorageFile storageFile;
-    private final AddressBook addressBook;
+    public void run(String[] launchArgs, InputStream inputStream, PrintStream outputStream) {
+        start(launchArgs, inputStream, outputStream);
+        runCommandLoopUntilExitCommand();
+        exit();
+    }
 
     /**
-     * Sets up the different components and loads up the model from the storage file.
+     * Sets up the different components, loads up the data from the storage file, and prints the welcome message.
      *
-     * @param storageFilePath file path of the desired storage file
+     * @param launchArgs arguments supplied by the user at program launch
      * @param inputStream user text input source
      * @param outputStream user text output acceptor
      *
-     * @throws MainInitialisationException if there were problems
      */
-    public Main(String storageFilePath, InputStream inputStream, PrintStream outputStream)
```

Comment:
Removed init code elsewhere because It is recommended that the constructor is kept as simple as possible.